core/src/scene/sceneLoader.cpp

```diff
@@ -944,29 +941,16 @@ void SceneLoader::loadSource(const std::shared_ptr<Platform>& platform, const st
         return;
     }
 
-    std::string type = source["type"].Scalar();
+    std::string type;
     std::string url;
     std::string mbtiles;
     int32_t minDisplayZoom = -1;
     int32_t maxDisplayZoom = -1;
     int32_t maxZoom = 18;
 
-    std::string mime;
-
-    if (type == "GeoJSON") {
-        mime = "application/geo+json";
-    } else if (type == "TopoJSON") {
-        mime =  "application/topo+json";
-    } else if (type == "MVT") {
-        mime =  "application/vnd.mapbox-vector-tile";
-    } else if (type == "Raster") {
-        // TODO try to guess from url
-        mime = "image/png";
-    } else {
-        LOGW("Unrecognized data source type '%s', skipping", type.c_str());
-        return;
+    if (auto typeNode = source["type"]) {
+        type = typeNode.Scalar();
     }
```

Comment:
Good point, we effectively default to the `GeoJSON` format if the string isn't recognized, but instead we should print a warning and not add the data source to the scene.